Kassiopeia/Interactions/Source/KSIntCalculatorMott.cxx

```diff
@@ -0,0 +1,222 @@
+    KThreeVector tPosition = anInitialParticle.GetPosition();
+    KThreeVector tMomentum = anInitialParticle.GetMomentum();
+
+    double tTheta = GetTheta(tInitialKineticEnergy);
+    double tLostKineticEnergy = GetEnergyLoss(tInitialKineticEnergy, tTheta);
+
+
+    double tPhi = KRandom::GetInstance().Uniform(0., 2. * katrin::KConst::Pi());
+
+    KThreeVector tOrthogonalOne = tMomentum.Orthogonal();
+    KThreeVector tOrthogonalTwo = tMomentum.Cross(tOrthogonalOne);
+    KThreeVector tFinalDirection =
+        tMomentum.Magnitude() *
+        (sin(tTheta* katrin::KConst::Pi() / 180) * (cos(tPhi) * tOrthogonalOne.Unit() + sin(tPhi) * tOrthogonalTwo.Unit()) +
+         cos(tTheta* katrin::KConst::Pi() / 180) * tMomentum.Unit());
+
+
+    aFinalParticle.SetTime(tTime);
+    aFinalParticle.SetPosition(tPosition);
+    aFinalParticle.SetMomentum(tFinalDirection);
+    aFinalParticle.SetKineticEnergy_eV(tInitialKineticEnergy - tLostKineticEnergy);
+    aFinalParticle.SetLabel(GetName());
+
+    fStepNInteractions = 1;
+    fStepEnergyLoss = tLostKineticEnergy;
+    fStepAngularChange = tTheta;
+
+    return;
+}
+
+double KSIntCalculatorMott::GetTheta(const double& anEnergy){
+
+    double tTheta;
+    double resolution;
+    double resolution_factor = 10; // arbitrarily chosen for o(1ms) computation time for He; computation time increases linearly with resolution factor; resolution factors under 1 yields incorrect results
+
+
+    resolution = ceil((fThetaMax - fThetaMin) * resolution_factor);
+
+
+    // Initializing array for possible theta values
+    std::vector<double> theta_arr;
+    for (int i = 0; i <= int(resolution); ++i) {
+        theta_arr.push_back(fThetaMin + i * (fThetaMax - fThetaMin) / resolution);
+    }
+
+    // Calculating rescaling factor for rejection sampling
+    std::vector<double> k_values;
+    for (double theta : theta_arr) {
+        k_values.push_back(MDCS(theta, anEnergy) / Normalized_RDCS(theta));
+    }
+
+    // Find the maximum value of k
+    double k = *std::max_element(k_values.begin(), k_values.end());
+
+
+
+    // Rejection sampling taking initial sample from Rutherford Diff. X-Sec
+    while (true) {
+        double sample = Normalized_RTCSInverse(KRandom::GetInstance().Uniform(0.0, 1.0, false, true));
+        double uniform_sample = KRandom::GetInstance().Uniform(0.0, 1.0, false, true);
+        if (MDCS(sample, anEnergy) / (k * Normalized_RDCS(sample)) > uniform_sample) {
+            tTheta = sample;
+            break;
+        }
+    }
+
+    return tTheta;
+
+}
+
+double KSIntCalculatorMott::GetEnergyLoss(const double& anEnergy, const double& aTheta){
+    double M, me, p, anELoss;
+
+    double amu = 0.0;
+
+    if (fNucleus == "He") {
+        amu = 4.0026; // mass in amu from W.J. Huang et al 2021 Chinese Phys. C 45 030002
+    } else if (fNucleus == "Ne") {
+        amu = 19.9924; // mass in amu from W.J. Huang et al 2021 Chinese Phys. C 45 030002
```

Comment:
This is for <sup>20</sup>Ne and the above for <sup>4</sup>He, right? Is your entire code specific to those isotopes?

Reply:
In that case I'm wondering if it makes sense to include the mass number in `fNucleus` so the configuration is clear and everything is easily extendible?

Reply:
Well, the cross-section is independent of mass number. It only comes in when calculating the electron energy loss. In any case, I've updated it to require mass number on the nucleus entry. I've added comments about where calculations are independent of mass number

Reply:
Thank you very much!